node/v2/header.js

```diff
@@ -69,11 +71,28 @@ HeaderRW.prototype.writeInto = function writeInto(headers, buffer, offset) {
 
     res = self.countrw.writeInto(keys.length, buffer, offset);
 
+    if (self.enforceHeaderCount && keys.length > 128) {
+        return bufrw.WriteResult.error(errors.TooManyHeaders({
+            offset: offset,
+            endOffset: res.offset,
+            count: keys.length
+        }), offset);
+    }
+
     for (var i = 0; i < keys.length; i++) {
         if (res.err) return res;
         offset = res.offset;
 
         var key = keys[i];
+        // TODO: Check that its' 16 bytes
+        if (self.enforceKeyLength && key.length > 16) {
+            return bufrw.WriteResult.error(errors.TransportHeaderTooLong({
+                offset: offset,
+                endOffset: res.offset,
+                headerName: key
+            }), offset);
+        }
+
         res = self.keyrw.writeInto(key, buffer, offset);
         if (res.err) return res;
         offset = res.offset;
```

Comment:
you can TO-DONE that by keeping a start offset, and then calculating the difference here.

Basically: just do the write, and then check the bytes here